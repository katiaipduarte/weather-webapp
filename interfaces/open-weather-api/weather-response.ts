import { CurrentWeather } from './current-weather'
import { DailyWeather } from './daily-weather'

export interface WeatherResponse {
	lat: number
	lon: number
	timezone: string
	timezone_offset: number
	current: CurrentWeather
	daily: DailyWeather[]
}
