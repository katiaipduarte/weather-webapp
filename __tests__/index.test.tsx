import { mockWeatherResponse } from '@mocks/weather-response.mock'
import Home from '@pages/index'
import { getWeatherByCoords } from '@services/weather.service'
import { render, screen } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import theme from '../styles/theme'

jest.mock('../services/weather.service')

describe('Home', () => {
	const renderComponent = ({ theme }) =>
		render(
			<ThemeProvider theme={theme}>
				<Home />
			</ThemeProvider>
		)

	const mockService = getWeatherByCoords as jest.MockedFunction<
		typeof getWeatherByCoords
	>

	test('renders homepage unchanged', async () => {
		mockService.mockResolvedValue(mockWeatherResponse)

		// Render new instance in every test to prevent leaking state
		const { container } = renderComponent({ theme: theme })

		expect(container).toMatchSnapshot()
	})

	test('renders a heading', () => {
		renderComponent({ theme: theme })

		const heading = screen.getByRole('heading', {
			name: 'My page',
		})

		expect(heading).toBeInTheDocument()
	})
})
