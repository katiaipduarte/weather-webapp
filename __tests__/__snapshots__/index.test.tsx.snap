// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders homepage unchanged 1`] = `
<div>
  <main
    style="background-image: url();"
  >
    <h1
      class="sc-gsnTZi hGIKZL logo"
    >
      weather.app‚Å¢
    </h1>
  </main>
  <footer />
</div>
`;
